app.go

```diff
@@ -197,6 +210,16 @@ func (a *App) RunAsSubcommand(ctx *Context) error {
 		}
 	}
 
+	if a.After != nil {
+		defer func() {
+			aferr := a.After(context)
+			// check Action error to avoid shadowing
+			if err == nil {
```

Comment:
Indeed, maybe with a comment that `err` is always `nil` there. I think we also need to change line 246 to `return err` to get it to propogate.